Snapshot & clone creation test on cStor volume.

gprasath opened the pull request:
Signed-off-by: gprasath <giridhara.prasad@cloudbyte.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Making snapshot/clone creation case applicable for cstor volume providing PVC name as input.
- Obtaining the storage class used from PVC spec.
- Obtaining the storage engine from volume spec and spc from the storage class spec.
- Checking if the snapshots are created using zfs list command based on the storage engine.


**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

vishnuitta commented on apps/percona/functional/k8s_snapshot/test.yml:
```diff
@@ -98,6 +114,30 @@
+              register: spc
+
+            - name: Obtaining the pool pods
+              shell: kubectl get pods -n {{ operator_ns }} | grep cstor-sparse-pool | awk '{print $1}'
```
there is hardcoded string here

vishnuitta commented on apps/percona/functional/k8s_snapshot/test.yml:
```diff
@@ -98,6 +114,30 @@
+              shell: kubectl get pvc {{ pvc_name }} -n {{ app_ns }} --no-headers -o custom-columns=:.spec.storageClassName
+              register: app_sc
+
+            - name: Obtaining the SPC from storage class
```
to avoid 'grep' as much as possible, probably, you can look at cstor_delete_and_verify_pool_deployment.yml which uses one 'grep' to find the pool pod from pv. It first finds the deployments from CVRs, by selecting the CVR related to pv. And, it does a grep to find the pool pod from its deployment.
Actually, we can use 'replicaset' to even avoid this grep to find pool pod from its deployment.
```
kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<pool_deployment>")].metadata.name}'
kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<rs_name_from_above>")].metadata.name}'
```
In above, we are finding the replicaset which is controlled by that deployment, and, then, finding the pod which is controlled by that replicaset.

vishnuitta replied:
probably, we may need to get the count also to check if there are multiple results (like one pod in terminating and another in containercreating etc) by using range like `-o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\.openebs\.io\/name}{"\n"}{end}'` and wc -l

vishnuitta replied:
Modified kubectl commands to print the output in new lines (in case of multiple results as mentioned above) to use wc -l:
```
kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<pool_deployment>")].metadata.name}{"\n"}'
kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="<rs_name_from_above>")].metadata.name}{"\n"}'
```
With this, we will be able to avoid grep

gprasath commented:
```PLAYBOOK: test.yml *************************************************************
1 plays in ./percona/functional/k8s_snapshot/test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
task path: /percona/functional/k8s_snapshot/test.yml:19
ok: [127.0.0.1]
META: ran handlers

TASK [Record test instance/run ID] *********************************************
task path: /percona/functional/k8s_snapshot/test.yml:29
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /percona/functional/k8s_snapshot/test.yml:33
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /percona/functional/k8s_snapshot/test.yml:40
changed: [127.0.0.1] => {"changed": true, "checksum": "165d3319b38a884427b62c59461c8af6f4a0f6a7", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "78c03493459fd66f30bbd4206444a743", "mode": "0644", "owner": "root", "size": 433, "src": "/root/.ansible/tmp/ansible-tmp-1542350444.61-11748616418435/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /percona/functional/k8s_snapshot/test.yml:51
changed: [127.0.0.1] => {"changed": true, "cmd": ["kubectl", "apply", "-f", "litmus-result.yaml"], "delta": "0:00:01.479231", "end": "2018-11-16 06:40:46.802541", "failed_when_result": false, "rc": 0, "start": "2018-11-16 06:40:45.323310","stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/create-k8s-snapshot-clone unchanged", "stdout_lines": ["litmusresult.litmus.io/create-k8s-snapshot-clone unchanged"]}
TASK [Check whether the snapshot specific storage class is created.] ***********
task path: /percona/functional/k8s_snapshot/test.yml:56
changed: [127.0.0.1] => {"changed": true, "cmd": ["kubectl", "get", "sc", "openebs-snapshot-promoter"], "delta": "0:00:00.959105", "end": "2018-11-16 06:40:47.969021", "rc": 0, "start": "2018-11-16 06:40:47.009916", "stderr": "", "stderr_lines": [], "stdout": "NAME                        PROVISIONER                                                AGE\nopenebs-snapshot-promoter   volumesnapshot.external-storage.k8s.io/snapshot-promoter   36d", "stdout_lines": ["NAME            PROVISIONER                                                AGE", "openebs-snapshot-promoter   volumesnapshot.external-storage.k8s.io/snapshot-promoter   36d"]}

TASK [Checking if the application is running.] *********************************
task path: /percona/functional/k8s_snapshot/test.yml:60
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n app-percona-ns --no-headers -l name=percona -o custom-columns=:status.phase", "delta": "0:00:00.959389", "end": "2018-11-16 06:40:49.142994", "rc": 0, "start": "2018-11-16 06:40:48.183605", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [Obtaining the application pod name.] *************************************
task path: /percona/functional/k8s_snapshot/test.yml:69
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n app-percona-ns -l name=percona --no-headers -o custom-columns=:metadata.name", "delta": "0:00:00.986103", "end": "2018-11-16 06:40:50.336603", "rc": 0, "start": "2018-11-16 06:40:49.350500", "stderr": "", "stderr_lines": [], "stdout": "percona-5957995575-q2fqj", "stdout_lines": ["percona-5957995575-q2fqj"]}

TASK [Recording application pod name in a variable.] ***************************
task path: /percona/functional/k8s_snapshot/test.yml:75
ok: [127.0.0.1] => {"ansible_facts": {"pod_name": "percona-5957995575-q2fqj"}, "changed": false}
TASK [Derive PV name from PVC] *************************************************
task path: /percona/functional/k8s_snapshot/test.yml:79
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc percona-mysql-claim -n app-percona-ns --no-headers -ocustom-columns=:spec.volumeName", "delta": "0:00:00.961915", "end": "2018-11-16 06:40:51.559654", "rc": 0, "start": "2018-11-16 06:40:50.597739", "stderr": "", "stderr_lines": [], "stdout": "pvc-fcf145c0-e957-11e8-9943-42010a800fea", "stdout_lines": ["pvc-fcf145c0-e957-11e8-9943-42010a800fea"]}

TASK [Obtaining the storage-engine name from PV.] ******************************
task path: /percona/functional/k8s_snapshot/test.yml:87
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pv pvc-fcf145c0-e957-11e8-9943-42010a800fea --no-headers -o jsonpath=\"{.metadata.annotations.openebs\\\\.io/cas-type}\"", "delta": "0:00:00.968792", "end": "2018-11-16 06:40:52.717532", "rc": 0, "start": "2018-11-16 06:40:51.748740", "stderr": "", "stderr_lines": [], "stdout": "cstor", "stdout_lines": ["cstor"]}

TASK [Write a test database into percona mysql] ********************************
task path: /percona/functional/k8s_snapshot/test.yml:95
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl exec -it percona-5957995575-q2fqj -n app-percona-ns -- mysql -uroot -pk8sDem0 -e \"create database cloud;\"\n kubectl exec -it percona-5957995575-q2fqj -n app-percona-ns -- mysql -uroot -pk8sDem0 -e \"create table platforms (Data VARCHAR(20));\" cloud\n kubectl exec -it percona-5957995575-q2fqj -n app-percona-ns -- mysql -uroot -pk8sDem0 -e \"insert into platforms (Data) VALUES ('aws');\" cloud\n kubectl exec -it percona-5957995575-q2fqj -n app-percona-ns -- mysql -uroot -pk8sDem0 -e \"flush tables with read lock;\"", "delta": "0:00:02.104083", "end": "2018-11-16 06:40:55.010447", "failed_when_result": false, "rc": 0, "start": "2018-11-16 06:40:52.906364", "stderr": "Unable to use a TTY - input is not a terminal or the right kind of file\nmysql: [Warning] Using a password on the command line interface can be insecure.\nERROR 1007 (HY000) at line 1: Can't create database 'cloud'; database exists\ncommand terminated with exit code 1\nUnable to use a TTY - input is not a terminal or the right kind of file\nmysql: [Warning] Using a password on the command line interface can be insecure.\nERROR 1050 (42S01) at line 1: Table 'platforms' already exists\ncommand terminated with exit code 1\nUnable to use a TTY - input is not a terminal or the right kind of file\nmysql: [Warning] Using a password on the command line interface can be insecure.\nUnable to use a TTY - input is not a terminal or the right kind of file\nmysql: [Warning] Using a password on the command line interface can be insecure.", "stderr_lines": ["Unable to use a TTY - input is not a terminal or the right kind of file", "mysql: [Warning] Using a password on the command line interface can be insecure.", "ERROR 1007 (HY000) at line 1: Can't create database'cloud'; database exists", "command terminated with exit code 1", "Unable to use a TTY - input is not a terminal or theright kind of file", "mysql: [Warning] Using a password on the command line interface can be insecure.", "ERROR 1050 (42S01) at line 1: Table 'platforms' already exists", "command terminated with exit code 1", "Unable to use a TTY - input is not a terminal or the right kind of file", "mysql: [Warning] Using a password on the command line interface can be insecure.", "Unable to use a TTY - input is not a terminal or the right kind of file", "mysql: [Warning] Using a password on the command line interface can be insecure."], "stdout": "", "stdout_lines": []}

TASK [Perform filesystem sync on percona pod before creating snapshot.] ********
task path: /percona/functional/k8s_snapshot/test.yml:106
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl exec percona-5957995575-q2fqj -n app-percona-ns -- bash -c \"sync;sync;sync\"", "delta": "0:00:01.150022", "end": "2018-11-16 06:40:56.351166", "rc": 0, "start": "2018-11-16 06:40:55.201144", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Creating snapshot.] ******************************************************
task path: /percona/functional/k8s_snapshot/test.yml:114
included: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml for 127.0.0.1

TASK [Update the snapshot template with the values provided.] ******************
task path: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml:5
changed: [127.0.0.1] => {"changed": true, "checksum": "93ff68b25a88147da65cf23a4f852fb1a251731d", "dest": "./snapshot.yml", "gid": 0, "group": "root", "md5sum": "77317e6d3b3ee939f1b59b6cd9495a70", "mode": "0644", "owner": "root", "size": 196, "src": "/root/.ansible/tmp/ansible-tmp-1542350456.48-57639523921777/source", "state": "file", "uid": 0}

TASK [Creating the volume snapshot] ********************************************
task path: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml:10
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl apply -f snapshot.yml", "delta": "0:00:01.055730", "end": "2018-11-16 06:40:57.932032", "rc": 0, "start": "2018-11-16 06:40:56.876302", "stderr": "", "stderr_lines": [], "stdout": "volumesnapshot.volumesnapshot.external-storage.k8s.io/snapshot00-percona created", "stdout_lines": ["volumesnapshot.volumesnapshot.external-storage.k8s.io/snapshot00-percona created"]}

TASK [Check if the snapshot is created successfully] ***************************
task path: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml:19
FAILED - RETRYING: Check if the snapshot is created successfully (15 retries left).
changed: [127.0.0.1] => {"attempts": 2, "changed": true, "cmd": "kubectl describe volumesnapshot snapshot00-percona -n app-percona-ns | grep Message", "delta": "0:00:00.921798", "end": "2018-11-16 06:41:30.205775", "rc": 0, "start": "2018-11-16 06:41:29.283977", "stderr": "", "stderr_lines": [], "stdout": "    Message:               Snapshot created successfully", "stdout_lines": ["    Message:               Snapshot created successfully"]}

TASK [Obtaining the storage class used from PVC] *******************************
task path: /percona/functional/k8s_snapshot/test.yml:119
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc percona-mysql-claim -n app-percona-ns --no-headers -ocustom-columns=:.spec.storageClassName", "delta": "0:00:00.930451", "end": "2018-11-16 06:41:31.353655", "rc": 0, "start": "2018-11-16 06:41:30.423204", "stderr": "", "stderr_lines": [], "stdout": "openebs-cstor-sparse", "stdout_lines": ["openebs-cstor-sparse"]}

TASK [Obtaining the SPC from storage class] ************************************
task path: /percona/functional/k8s_snapshot/test.yml:123
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get sc openebs-cstor-sparse --no-headers -o yaml | awk '/StoragePoolClaim/{getline; print}' | cut -d ':' -f2 | sed 's/\"//g'", "delta": "0:00:00.902783", "end": "2018-11-16 06:41:32.450055", "rc": 0, "start": "2018-11-16 06:41:31.547272", "stderr": "", "stderr_lines": [], "stdout": " cstor-sparse-pool", "stdout_lines": [" cstor-sparse-pool"]}

TASK [Obtaining the pool deployments from cvr] *********************************
task path: /percona/functional/k8s_snapshot/test.yml:127
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get cvr -n openebs -l openebs.io/persistent-volume=pvc-fcf145c0-e957-11e8-9943-42010a800fea --no-headers -o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\\.openebs\\.io\\/name}{\"\\n\"}{end}'", "delta": "0:00:00.912861", "end": "2018-11-16 06:41:33.539403", "rc": 0, "start": "2018-11-16 06:41:32.626542", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe\ncstor-sparse-pool-h8wm\ncstor-sparse-pool-qi1e", "stdout_lines": ["cstor-sparse-pool-9joe", "cstor-sparse-pool-h8wm", "cstor-sparse-pool-qi1e"]}
TASK [Obtaining the replicasets corresponding to pool deployements.] ***********
task path: /percona/functional/k8s_snapshot/test.yml:136
changed: [127.0.0.1] => (item=cstor-sparse-pool-9joe) => {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe\")].metadata.name}'", "delta": "0:00:00.921580", "end": "2018-11-16 06:41:34.651140", "item": "cstor-sparse-pool-9joe", "rc": 0, "start": "2018-11-16 06:41:33.729560", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe-5f86dd47c6", "stdout_lines": ["cstor-sparse-pool-9joe-5f86dd47c6"]}
changed: [127.0.0.1] => (item=cstor-sparse-pool-h8wm) => {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm\")].metadata.name}'", "delta": "0:00:00.911939", "end": "2018-11-16 06:41:35.702248", "item": "cstor-sparse-pool-h8wm", "rc": 0, "start": "2018-11-16 06:41:34.790309", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h8wm-fdcb888c8","stdout_lines": ["cstor-sparse-pool-h8wm-fdcb888c8"]}
changed: [127.0.0.1] => (item=cstor-sparse-pool-qi1e) => {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e\")].metadata.name}'", "delta": "0:00:00.921590", "end": "2018-11-16 06:41:36.778772", "item": "cstor-sparse-pool-qi1e", "rc": 0, "start": "2018-11-16 06:41:35.857182", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-qi1e-9d56bf87f","stdout_lines": ["cstor-sparse-pool-qi1e-9d56bf87f"]}
TASK [Obtaining the pool pods] *************************************************
task path: /percona/functional/k8s_snapshot/test.yml:144
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-9joe-5f86dd47c6', '_ansible_item_result': True, u'delta': u'0:00:00.921580', 'stdout_lines': [u'cstor-sparse-pool-9joe-5f86dd47c6'], '_ansible_item_label': u'cstor-sparse-pool-9joe', u'end': u'2018-11-16 06:41:34.651140', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', 'item': u'cstor-sparse-pool-9joe', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:33.729560', '_ansible_ignore_errors': None}) => {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe-5f86dd47c6\")].metadata.name}'", "delta": "0:00:00.956267", "end": "2018-11-16 06:41:37.933274", "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe\")].metadata.name}'", "delta": "0:00:00.921580", "end": "2018-11-16 06:41:34.651140", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-9joe", "rc": 0, "start": "2018-11-16 06:41:33.729560", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe-5f86dd47c6", "stdout_lines": ["cstor-sparse-pool-9joe-5f86dd47c6"]}, "rc": 0, "start": "2018-11-16 06:41:36.977007", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe-5f86dd47c6-ttphm", "stdout_lines": ["cstor-sparse-pool-9joe-5f86dd47c6-ttphm"]}
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-h8wm-fdcb888c8', '_ansible_item_result': True, u'delta': u'0:00:00.911939', 'stdout_lines': [u'cstor-sparse-pool-h8wm-fdcb888c8'], '_ansible_item_label': u'cstor-sparse-pool-h8wm', u'end': u'2018-11-16 06:41:35.702248', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', 'item': u'cstor-sparse-pool-h8wm', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:34.790309', '_ansible_ignore_errors': None}) => {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm-fdcb888c8\")].metadata.name}'", "delta": "0:00:00.897223", "end": "2018-11-16 06:41:38.982660", "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm\")].metadata.name}'", "delta": "0:00:00.911939", "end": "2018-11-16 06:41:35.702248", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-h8wm", "rc": 0, "start": "2018-11-16 06:41:34.790309", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h8wm-fdcb888c8", "stdout_lines": ["cstor-sparse-pool-h8wm-fdcb888c8"]}, "rc": 0, "start": "2018-11-16 06:41:38.085437", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h8wm-fdcb888c8-7lppr", "stdout_lines": ["cstor-sparse-pool-h8wm-fdcb888c8-7lppr"]}
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-qi1e-9d56bf87f', '_ansible_item_result': True, u'delta': u'0:00:00.921590', 'stdout_lines': [u'cstor-sparse-pool-qi1e-9d56bf87f'], '_ansible_item_label': u'cstor-sparse-pool-qi1e', u'end': u'2018-11-16 06:41:36.778772', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', 'item': u'cstor-sparse-pool-qi1e', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:35.857182', '_ansible_ignore_errors': None}) => {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e-9d56bf87f\")].metadata.name}'", "delta": "0:00:00.955581", "end": "2018-11-16 06:41:40.079000", "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e\")].metadata.name}'", "delta": "0:00:00.921590", "end": "2018-11-16 06:41:36.778772", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-qi1e", "rc": 0, "start": "2018-11-16 06:41:35.857182", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-qi1e-9d56bf87f", "stdout_lines": ["cstor-sparse-pool-qi1e-9d56bf87f"]}, "rc": 0, "start": "2018-11-16 06:41:39.123419", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-qi1e-9d56bf87f-nxfn4", "stdout_lines": ["cstor-sparse-pool-qi1e-9d56bf87f-nxfn4"]}
TASK [Checking if the snapshot is created in all the pool pods.] ***************
task path: /percona/functional/k8s_snapshot/test.yml:152
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-9joe-5f86dd47c6-ttphm', '_ansible_item_result': True, u'delta': u'0:00:00.956267', 'stdout_lines': [u'cstor-sparse-pool-9joe-5f86dd47c6-ttphm'], '_ansible_item_label': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-9joe-5f86dd47c6', '_ansible_item_result': True, u'delta': u'0:00:00.921580', 'stdout_lines': [u'cstor-sparse-pool-9joe-5f86dd47c6'], '_ansible_item_label': u'cstor-sparse-pool-9joe', u'end': u'2018-11-16 06:41:34.651140', '_ansible_no_log': False, u'start': u'2018-11-16 06:41:33.729560', u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', 'failed': False, 'item': u'cstor-sparse-pool-9joe', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'end': u'2018-11-16 06:41:37.933274', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe-5f86dd47c6")].metadata.name}\'', 'item': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-9joe-5f86dd47c6', '_ansible_item_result': True, u'delta': u'0:00:00.921580', 'stdout_lines': [u'cstor-sparse-pool-9joe-5f86dd47c6'], '_ansible_item_label': u'cstor-sparse-pool-9joe', u'end': u'2018-11-16 06:41:34.651140', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', u'start': u'2018-11-16 06:41:33.729560', 'item': u'cstor-sparse-pool-9joe', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-9joe-5f86dd47c6")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:36.977007', '_ansible_ignore_errors': None}) => {"changed": true, "cmd": "kubectl exec -ti cstor-sparse-pool-9joe-5f86dd47c6-ttphm -n openebs -- zfs list -t snapshot", "delta": "0:00:01.140922", "end": "2018-11-16 06:41:41.417017", "failed_when_result": false, "item": {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe-5f86dd47c6\")].metadata.name}'", "delta": "0:00:00.956267", "end": "2018-11-16 06:41:37.933274", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe-5f86dd47c6\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable":null, "removes": null, "stdin": null, "warn": true}}, "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe\")].metadata.name}'", "delta": "0:00:00.921580", "end": "2018-11-16 06:41:34.651140", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-9joe\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-9joe", "rc": 0, "start": "2018-11-16 06:41:33.729560", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe-5f86dd47c6", "stdout_lines": ["cstor-sparse-pool-9joe-5f86dd47c6"]}, "rc": 0, "start": "2018-11-16 06:41:36.977007","stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-9joe-5f86dd47c6-ttphm", "stdout_lines": ["cstor-sparse-pool-9joe-5f86dd47c6-ttphm"]}, "rc": 0, "start": "2018-11-16 06:41:40.276095", "stderr": "Defaulting container name to cstor-pool.\nUse 'kubectl describe pod/cstor-sparse-pool-9joe-5f86dd47c6-ttphm -n openebs' to see all of the containers inthis pod.\nUnable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Defaulting container name to cstor-pool.", "Use 'kubectl describe pod/cstor-sparse-pool-9joe-5f86dd47c6-ttphm -n openebs' to see all of the containers in this pod.", "Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": "NAME                                           USED  AVAIL  REFER  MOUNTPOINT\ncstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213178K      -  15.6M  -\ncstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447     177K      -  15.7M  -\ncstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_1542350458018959715    60K      -  15.7M  -", "stdout_lines": ["NAME                                                                                                          USED  AVAIL  REFER  MOUNTPOINT", "cstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213         178K      -  15.6M  -", "cstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447     177K      -  15.7M  -", "cstor-93328696-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_1542350458018959715    60K      -  15.7M  -"]}
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-h8wm-fdcb888c8-7lppr', '_ansible_item_result': True, u'delta': u'0:00:00.897223', 'stdout_lines': [u'cstor-sparse-pool-h8wm-fdcb888c8-7lppr'], '_ansible_item_label': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-h8wm-fdcb888c8', '_ansible_item_result': True, u'delta': u'0:00:00.911939', 'stdout_lines': [u'cstor-sparse-pool-h8wm-fdcb888c8'], '_ansible_item_label': u'cstor-sparse-pool-h8wm', u'end': u'2018-11-16 06:41:35.702248', '_ansible_no_log': False, u'start': u'2018-11-16 06:41:34.790309', u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', 'failed': False, 'item': u'cstor-sparse-pool-h8wm', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'end': u'2018-11-16 06:41:38.982660', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm-fdcb888c8")].metadata.name}\'', 'item': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-h8wm-fdcb888c8', '_ansible_item_result': True, u'delta': u'0:00:00.911939', 'stdout_lines': [u'cstor-sparse-pool-h8wm-fdcb888c8'], '_ansible_item_label': u'cstor-sparse-pool-h8wm', u'end': u'2018-11-16 06:41:35.702248', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -nopenebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', u'start': u'2018-11-16 06:41:34.790309', 'item': u'cstor-sparse-pool-h8wm', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-h8wm-fdcb888c8")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:38.085437', '_ansible_ignore_errors':None}) => {"changed": true, "cmd": "kubectl exec -ti cstor-sparse-pool-h8wm-fdcb888c8-7lppr -n openebs -- zfs list -t snapshot", "delta": "0:00:01.111468", "end": "2018-11-16 06:41:42.674135", "failed_when_result": false, "item": {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm-fdcb888c8\")].metadata.name}'", "delta": "0:00:00.897223", "end": "2018-11-16 06:41:38.982660", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm-fdcb888c8\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm\")].metadata.name}'", "delta": "0:00:00.911939", "end": "2018-11-16 06:41:35.702248", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-h8wm\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-h8wm", "rc": 0, "start": "2018-11-16 06:41:34.790309", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h8wm-fdcb888c8", "stdout_lines": ["cstor-sparse-pool-h8wm-fdcb888c8"]}, "rc": 0, "start": "2018-11-16 06:41:38.085437", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h8wm-fdcb888c8-7lppr", "stdout_lines": ["cstor-sparse-pool-h8wm-fdcb888c8-7lppr"]}, "rc": 0, "start": "2018-11-16 06:41:41.562667", "stderr": "Defaulting container name to cstor-pool.\nUse 'kubectl describe pod/cstor-sparse-pool-h8wm-fdcb888c8-7lppr -n openebs' to see all of the containers in this pod.\nUnable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Defaulting container name to cstor-pool.", "Use 'kubectl describe pod/cstor-sparse-pool-h8wm-fdcb888c8-7lppr -n openebs' to see all of the containers inthis pod.", "Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": "NAME                          USED  AVAIL  REFER  MOUNTPOINT\ncstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213         181K      -  15.6M  -\ncstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447     180K      -  15.7M  -\ncstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_154235045801895971561.5K      -  15.7M  -", "stdout_lines": ["NAME                                                                                         USED  AVAIL  REFER  MOUNTPOINT", "cstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213         181K      -  15.6M  -", "cstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447180K      -  15.7M  -", "cstor-9330634a-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_1542350458018959715  61.5K      -  15.7M  -"]}
changed: [127.0.0.1] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stdout': u'cstor-sparse-pool-qi1e-9d56bf87f-nxfn4', '_ansible_item_result': True, u'delta': u'0:00:00.955581', 'stdout_lines': [u'cstor-sparse-pool-qi1e-9d56bf87f-nxfn4'], '_ansible_item_label': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-qi1e-9d56bf87f', '_ansible_item_result': True, u'delta': u'0:00:00.921590', 'stdout_lines': [u'cstor-sparse-pool-qi1e-9d56bf87f'], '_ansible_item_label': u'cstor-sparse-pool-qi1e', u'end': u'2018-11-16 06:41:36.778772', '_ansible_no_log': False, u'start': u'2018-11-16 06:41:35.857182', u'cmd': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', 'failed': False, 'item': u'cstor-sparse-pool-qi1e', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'end': u'2018-11-16 06:41:40.079000', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e-9d56bf87f")].metadata.name}\'', 'item': {'_ansible_parsed': True, 'stderr_lines': [], u'changed': True, u'stderr': u'', u'stdout': u'cstor-sparse-pool-qi1e-9d56bf87f', '_ansible_item_result': True, u'delta': u'0:00:00.921590', 'stdout_lines': [u'cstor-sparse-pool-qi1e-9d56bf87f'], '_ansible_item_label': u'cstor-sparse-pool-qi1e', u'end': u'2018-11-16 06:41:36.778772', '_ansible_no_log': False, 'failed': False, u'cmd': u'kubectl get rs --selector=app=cstor-pool -nopenebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', u'start': u'2018-11-16 06:41:35.857182', 'item': u'cstor-sparse-pool-qi1e', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, '_ansible_ignore_errors': None}, u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u'kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath=\'{.items[?(@.metadata.ownerReferences[0].name=="cstor-sparse-pool-qi1e-9d56bf87f")].metadata.name}\'', u'removes': None, u'argv': None, u'creates': None, u'chdir': None, u'stdin': None}}, u'start': u'2018-11-16 06:41:39.123419', '_ansible_ignore_errors':None}) => {"changed": true, "cmd": "kubectl exec -ti cstor-sparse-pool-qi1e-9d56bf87f-nxfn4 -n openebs -- zfs list -t snapshot", "delta": "0:00:01.264434", "end": "2018-11-16 06:41:44.098385", "failed_when_result": false, "item": {"changed": true, "cmd": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e-9d56bf87f\")].metadata.name}'", "delta": "0:00:00.955581", "end": "2018-11-16 06:41:40.079000", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get pod --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e-9d56bf87f\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": {"changed": true, "cmd": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e\")].metadata.name}'", "delta": "0:00:00.921590", "end": "2018-11-16 06:41:36.778772", "failed": false, "invocation": {"module_args": {"_raw_params": "kubectl get rs --selector=app=cstor-pool -n openebs --no-headers -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name==\"cstor-sparse-pool-qi1e\")].metadata.name}'", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": null, "removes": null, "stdin": null, "warn": true}}, "item": "cstor-sparse-pool-qi1e", "rc": 0, "start": "2018-11-16 06:41:35.857182", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-qi1e-9d56bf87f", "stdout_lines": ["cstor-sparse-pool-qi1e-9d56bf87f"]}, "rc": 0, "start": "2018-11-16 06:41:39.123419", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-qi1e-9d56bf87f-nxfn4", "stdout_lines": ["cstor-sparse-pool-qi1e-9d56bf87f-nxfn4"]}, "rc": 0, "start": "2018-11-16 06:41:42.833951", "stderr": "Defaulting container name to cstor-pool.\nUse 'kubectl describe pod/cstor-sparse-pool-qi1e-9d56bf87f-nxfn4 -n openebs' to see all of the containers in this pod.\nUnable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Defaulting container name to cstor-pool.", "Use 'kubectl describe pod/cstor-sparse-pool-qi1e-9d56bf87f-nxfn4 -n openebs' to see all of the containers inthis pod.", "Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": "NAME                          USED  AVAIL  REFER  MOUNTPOINT\ncstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213         178K      -  15.6M  -\ncstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447     178K      -  15.7M  -\ncstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_154235045801895971560.5K      -  15.7M  -", "stdout_lines": ["NAME                                                                                         USED  AVAIL  REFER  MOUNTPOINT", "cstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snap-percona_1542348698769828213         178K      -  15.6M  -", "cstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot-percona_1542349554514462447178K      -  15.7M  -", "cstor-9331b37d-e957-11e8-9943-42010a800fea/pvc-fcf145c0-e957-11e8-9943-42010a800fea@pvc-fcf145c0-e957-11e8-9943-42010a800fea_snapshot00-percona_1542350458018959715  60.5K      -  15.7M  -"]}

TASK [Creating clone using snapshot.] ******************************************
task path: /percona/functional/k8s_snapshot/test.yml:164
included: /funclib/kubectl/k8s_snapshot_clone/create_clone.yml for 127.0.0.1

TASK [Update the clone creation template with the values provided.] ************
task path: /funclib/kubectl/k8s_snapshot_clone/create_clone.yml:5
changed: [127.0.0.1] => {"changed": true, "checksum": "19280ac3495ba90f636e2d30a1475c115bd1581e", "dest": "./snapshot-claim.yml", "gid": 0, "group": "root", "md5sum": "8729235eda8a1a725bd3341ea06dedff", "mode": "0644", "owner": "root", "size": 316, "src": "/root/.ansible/tmp/ansible-tmp-1542350504.24-224582443073625/source", "state": "file", "uid": 0}

TASK [Creating PVC from the snapshot] ******************************************
task path: /funclib/kubectl/k8s_snapshot_clone/create_clone.yml:10
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f snapshot-claim.yml", "delta": "0:00:01.099769", "end": "2018-11-16 06:41:45.729084", "rc": 0, "start": "2018-11-16 06:41:44.629315", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim/snap-mysql-claim created", "stdout_lines": ["persistentvolumeclaim/snap-mysql-claim created"]}

TASK [Checking if the pvc is created successfully] *****************************
task path: /funclib/kubectl/k8s_snapshot_clone/create_clone.yml:15
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pvc -n app-percona-ns | grep snap-mysql-claim", "delta": "0:00:00.940240", "end": "2018-11-16 06:41:46.909177", "rc": 0, "start": "2018-11-16 06:41:45.968937", "stderr": "", "stderr_lines": [], "stdout": "snap-mysql-claim      Bound    pvc-afa1803f-e96a-11e8-9943-42010a800fea   5G       RWO            openebs-snapshot-promoter   1s", "stdout_lines": ["snap-mysql-claim      Bound    pvc-afa1803f-e96a-11e8-9943-42010a800fea   5G         RWO            openebs-snapshot-promoter   1s"]}

TASK [Obtaining new PV name from the pvc.] *************************************
task path: /percona/functional/k8s_snapshot/test.yml:167
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc snap-mysql-claim -n app-percona-ns -o custom-columns=:spec.volumeName --no-headers", "delta": "0:00:00.893362", "end": "2018-11-16 06:41:48.035598", "rc": 0, "start": "2018-11-16 06:41:47.142236", "stderr": "", "stderr_lines": [], "stdout": "pvc-afa1803f-e96a-11e8-9943-42010a800fea", "stdout_lines": ["pvc-afa1803f-e96a-11e8-9943-42010a800fea"]}

TASK [List the corresponding storage replica pods..] ***************************
task path: /percona/functional/k8s_snapshot/test.yml:173
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n app-percona-ns -l openebs.io/replica=jiva-replica| grep pvc-afa1803f-e96a-11e8-9943-42010a800fea | awk '{print $1}'", "delta": "0:00:00.901740", "end": "2018-11-16 06:41:49.122527", "rc": 0, "start": "2018-11-16 06:41:48.220787", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [check if the sync is completed in all the above replicas.] ***************
task path: /percona/functional/k8s_snapshot/test.yml:179

TASK [Update the percona deployment spec with the clone pvc.] ******************
task path: /percona/functional/k8s_snapshot/test.yml:190
changed: [127.0.0.1] => {"changed": true, "checksum": "f2e2f192e56175b50efcb5922b837d29acf3b05a", "dest": "./percona.yml", "gid": 0, "group": "root", "md5sum": "82e67a2f6906427a4642962f3d168aac", "mode": "0644", "owner": "root", "size": 853, "src": "/root/.ansible/tmp/ansible-tmp-1542350509.24-129179365290593/source", "state": "file", "uid": 0}

TASK [Deploy the application using snapped volume.] ****************************
task path: /percona/functional/k8s_snapshot/test.yml:195
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f percona.yml -n app-percona-ns", "delta": "0:00:01.044479", "end": "2018-11-16 06:41:50.688257", "rc": 0, "start": "2018-11-16 06:41:49.643778", "stderr": "", "stderr_lines": [], "stdout": "deployment.apps/percona-new created", "stdout_lines": ["deployment.apps/percona-new created"]}

TASK [Check if the new application pod is started.] ****************************
task path: /percona/functional/k8s_snapshot/test.yml:200
FAILED - RETRYING: Check if the new application pod is started. (10 retries left).
FAILED - RETRYING: Check if the new application pod is started. (9 retries left).
FAILED - RETRYING: Check if the new application pod is started. (8 retries left).
changed: [127.0.0.1] => {"attempts": 4, "changed": true, "cmd": "kubectl get pods -n app-percona-ns -l name=percona-new", "delta": "0:00:00.899869", "end": "2018-11-16 06:44:55.098818", "rc": 0, "start": "2018-11-16 06:44:54.198949", "stderr": "", "stderr_lines": [], "stdout": "NAME                           READY   STATUS    RESTARTS   AGE\npercona-new-56b8b557d8-5j4zv   1/1     Running   0          3m5s", "stdout_lines": ["NAME                           READY   STATUS    RESTARTS   AGE", "percona-new-56b8b557d8-5j4zv   1/1     Running   0          3m5s"]}

TASK [Obtaining the application pod name.] *************************************
task path: /percona/functional/k8s_snapshot/test.yml:209
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n app-percona-ns -l name=percona-new --no-headers -o custom-columns=:metadata.name", "delta": "0:00:00.891125", "end": "2018-11-16 06:44:56.182424", "rc": 0, "start": "2018-11-16 06:44:55.291299", "stderr": "", "stderr_lines": [], "stdout": "percona-new-56b8b557d8-5j4zv", "stdout_lines": ["percona-new-56b8b557d8-5j4zv"]}

TASK [Validate the data in the new application pod db.] ************************
task path: /percona/functional/k8s_snapshot/test.yml:215
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl exec -it percona-new-56b8b557d8-5j4zv -n app-percona-ns -- mysql -uroot -pk8sDem0 -e \"select * from platforms;\" cloud;", "delta": "0:00:01.112878", "end": "2018-11-16 06:44:57.497280", "rc": 0, "start": "2018-11-16 06:44:56.384402", "stderr": "Unable to use a TTY - input is not a terminal or the right kind of file\nmysql: [Warning] Using a password on the command line interface can be insecure.", "stderr_lines": ["Unable to use a TTY - input is not a terminal or the right kind of file", "mysql: [Warning] Using a password on the command line interface can be insecure."], "stdout": "Data\naws\naws\naws\naws", "stdout_lines": ["Data", "aws", "aws", "aws", "aws"]}

TASK [set_fact] ****************************************************************
task path: /percona/functional/k8s_snapshot/test.yml:224
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /percona/functional/k8s_snapshot/test.yml:233
changed: [127.0.0.1] => {"changed": true, "checksum": "86e9a8e1c5b8b82c3da3ae4f88c14234a53818f7", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "a3b5ff0c0471589cec3188a6311b1ab7", "mode": "0644", "owner": "root", "size": 431, "src": "/root/.ansible/tmp/ansible-tmp-1542350697.64-273552641032371/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /percona/functional/k8s_snapshot/test.yml:244
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.110070", "end": "2018-11-16 06:45:00.804047", "failed_when_result": false, "rc": 0, "start": "2018-11-16 06:44:59.693977", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/create-k8s-snapshot-clone configured", "stdout_lines": ["litmusresult.litmus.io/create-k8s-snapshot-clone configured"]}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=35   changed=30   unreachable=0    failed=0```
